### PR TITLE
Check for updates when user logs in

### PR DESCRIPTION
**Pull request by lucianogorza**
### Description
Every time the user logins, the frontend should make a request to the Wazuh API to check for new updates and display notifications if applicable.
Additionally, to determine which user is trying to get and set preferences, a security factory was added to the Wazhu Core plugin.

### Issues Resolved
- https://github.com/wazuh/wazuh-dashboard/issues/109

#### After login
image
image

#### After refresh the page
image

#### After close the notification
image

### Test
|                 |                                                                      |
|-----------------|----------------------------------------------------------------------|
| :black_circle:  | The test hasn't started yet.                                         |
| :green_circle:  | All checks passed.                                                   |
| :red_circle:    | There is at least one failed check.                                  |
| :yellow_circle: | There is at least one expected fail or skipped test and no failures. |

| Test | Chrome | Firefox | Safari
|--|--|--|--|
| "checkUpdates" doesn't appear in Session Storage before login |  |  |  |
| "checkUpdates" appears with value "executed" in Session Storage after login |  |  |  |
| "checkUpdates" doesn't appear in Session Storage after logout |  |  |  |
| After user logins there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=true" |  |  |  |
| After refresh the page there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=false" |  |  |  |
| Close the notification, refresh the page. The notification shouldn't appear |  |  |  |
| Check if the saved object "wazuh-check-updates-user-preferences:[username]" is correct |  |  |  |

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by asteriscos** [comment]
### Test
|                 |                                                                      |
|-----------------|----------------------------------------------------------------------|
| :white_circle:  | Skipped                                       |
| :black_circle:  | The test hasn't started yet.                                         |
| :green_circle:  | All checks passed.                                                   |
| :red_circle:    | There is at least one failed check.                                  |
| :yellow_circle: | There is at least one expected fail or skipped test and no failures. |

| Test | Chrome | Firefox | Safari
|--|--|--|--|
| "checkUpdates" doesn't appear in Session Storage before login | :green_circle: | :green_circle: | :white_circle: |
| "checkUpdates" appears with value "executed" in Session Storage after login | :green_circle: | :green_circle: |  :white_circle: |
| "checkUpdates" doesn't appear in Session Storage after logout | :green_circle: | :green_circle: | :white_circle: |
| After user logins there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=true" | :green_circle: | :green_circle: | :white_circle: |
| After refresh the page there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=false" | :green_circle: | :green_circle: | :white_circle: |
| Close the notification, refresh the page. The notification shouldn't appear |  |  | :white_circle: |
| Check if the saved object "wazuh-check-updates-user-preferences:[username]" is correct |  |  | :white_circle: |

**Comment by Tostti** [comment]
### Test
|                 |                                                                      |
|-----------------|----------------------------------------------------------------------|
| :white_circle:  | Skipped                                       |
| :black_circle:  | The test hasn't started yet.                                         |
| :green_circle:  | All checks passed.                                                   |
| :red_circle:    | There is at least one failed check.                                  |
| :yellow_circle: | There is at least one expected fail or skipped test and no failures. |

| Test | Chrome | Firefox | Safari
|--|--|--|--|
| "checkUpdates" doesn't appear in Session Storage before login | :green_circle: | :green_circle: | :white_circle: |
| "checkUpdates" appears with value "executed" in Session Storage after login | :green_circle: | :green_circle: |  :white_circle: |
| "checkUpdates" doesn't appear in Session Storage after logout | :green_circle: | :green_circle: | :white_circle: |
| After user logins there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=true" | :green_circle: | :green_circle: | :white_circle: |
| After refresh the page there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=false" | :green_circle: | :green_circle: | :white_circle: |
| Close the notification, refresh the page. The notification shouldn't appear |  :green_circle:   |   :green_circle:  | :white_circle: |
| Check if the saved object "wazuh-check-updates-user-preferences:[username]" is correct | :white_circle:  | :white_circle:  | :white_circle: |

<details>
<summary> "checkUpdates" doesn't appear in Session Storage before login </summary>

Chrome:
image

Firefox:
image

</details>

<details>
<summary> "checkUpdates" appears with value "executed" in Session Storage after login </summary>

Chrome:
image

Firefox:
image

</details>

<details>
<summary> "checkUpdates" doesn't appear in Session Storage after logout   </summary>

Chrome:
image

Firefox:
image

</details>

<details>
<summary> After user logins there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=true" </summary>

Chrome:
image

Firefox:
image


</details>

<details>
<summary> After refresh the page there is a network request "/api/wazuh-check-updates/updates?checkAvailableUpdates=false" </summary>

Chrome:
image

Firefox:
image

</details>

<details>
<summary>Close the notification, refresh the page. The notification shouldn't appear </summary>

Chrome:
image
image

Firefox:
image
image

</details>

<details>
<summary> Check if the saved object "wazuh-check-updates-user-preferences:[username]" is correct </summary>

Chrome:

Firefox:

</details>

**Review by Tostti — APPROVED** [review]
TEST:✔️
CR:✔️

LGTM

File: plugins/wazuh-check-updates/public/components/apis-update-status/table/columns/index.tsx [hunk 1]
```diff
@@ -28,6 +28,9 @@ export const getApisUpdateStatusColumns = () => {
           if (status === API_UPDATES_STATUS.AVAILABLE_UPDATES) {
```
[the remark this answers is not in the document]
Comment (lucianogorza) [reply plugins/wazuh-check-updates/public/components/apis-update-status/table/columns/index.tsx]: Changed!

**Review by Tostti — APPROVED** [review]

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark: 
Test: :heavy_check_mark:

**Pull request merged**